File: apps/api/src/main/java/br/com/agendaplatform/reporting/application/DashboardSummary.java
package br.com.agendaplatform.reporting.application;

import br.com.agendaplatform.availability.BlockSummary;
import br.com.agendaplatform.scheduling.AppointmentSummary;
import java.util.List;

public record DashboardSummary(
        List<AppointmentSummary> todayAppointments,
        AppointmentSummary nextAppointment,
        List<BlockSummary> todayBlocks,
        WeekSummary week) {
}

File: apps/admin/src/app/core/reporting/dashboard-summary.ts
import { AppointmentSummary } from '../scheduling/appointment-summary';
import { BlockSummary } from '../availability/block-summary';

export interface WeekSummary {
  readonly scheduledCount: number;
  readonly cancelledCount: number;
}

export interface DashboardSummary {
  readonly todayAppointments: AppointmentSummary[];
  readonly nextAppointment: AppointmentSummary | null;
  readonly todayBlocks: BlockSummary[];
  readonly week: WeekSummary;
}
